vignettes/adsl.Rmd

```diff
@@ -241,69 +241,59 @@ is to be imputed to the first of the month, then set `date_imputation = "FIRST"`
 
 ### Disposition Status (e.g. `EOSSTT`) {#disposition_status}
 
-The function `derive_var_disposition_status()` can be used to derive a disposition status at a specific
+The function `derive_var_merged_cat()` can be used to derive a disposition status at a specific
 timepoint. The relevant disposition variable (`DS.DSDECOD`) is selected by adjusting the filter 
 parameter and used to derive `EOSSTT`.
 
-To derive the End of Study status (`EOSSTT`), a call could be:
+To derive the End of Study status (`EOSSTT`), the function expects a mapping derivation for the `cat_fun` argument.
+The mapping derivation for the call below is
 
-```{r eval=TRUE}
-adsl <- adsl %>%
-  derive_var_disposition_status(
-    dataset_ds = ds,
-    new_var = EOSSTT,
-    status_var = DSDECOD,
-    filter_ds = DSCAT == "DISPOSITION EVENT"
-  )
-```
-
-```{r, eval=TRUE, echo=FALSE}
-dataset_vignette(adsl, display_vars = exprs(USUBJID, EOSDT, EOSSTT))
-```
-[Link](#link_ds) to `DS`.
-
-This call would return the input dataset with the column `EOSSTT` added. 
-
-By default, the function will derive `EOSSTT` as
-
-- `"NOT STARTED"` if `DSDECOD` is `"SCREEN FAILURE"` or `"SCREENING NOT COMPLETED"`
+- `"NOT STARTED"` if `DSDECOD` is `"SCREEN FAILURE"`
 - `"COMPLETED"` if `DSDECOD == "COMPLETED"`
 - `"DISCONTINUED"` if `DSDECOD` is not `"COMPLETED"` or `NA`
 - `"ONGOING"` otherwise
 
-If the default derivation must be changed, the user can create his/her own function and pass it to 
-the `format_new_var` argument of the function (`format_new_var = new_mapping`) to map `DSDECOD` 
-to a suitable `EOSSTT` value.
-
 Example function `format_eosstt()`:
 
 ```{r eval=TRUE}
-format_eosstt <- function(DSDECOD) {
+format_eosstt <- function(x) {
   case_when(
-    DSDECOD %in% c("COMPLETED") ~ "COMPLETED",
-    DSDECOD %in% c("SCREEN FAILURE") ~ NA_character_,
-    !is.na(DSDECOD) ~ "DISCONTINUED",
+    x %in% c("COMPLETED") ~ "COMPLETED",
+    x %in% c("SCREEN FAILURE") ~ NA_character_,
+    !is.na(x) ~ "DISCONTINUED",
     TRUE ~ "ONGOING"
   )
 }
 ```
 
 The customized mapping function `format_eosstt()` can now be passed to the main function:
 
-```{r eval=FALSE}
+a call could be:
```

Comment:
![image](https://user-images.githubusercontent.com/10111024/218544080-4a6ad069-0ba1-407f-aeac-163773cc1f29.png)

I don't think this is needed